app/cloud.go

```diff
@@ -4,14 +4,106 @@
 package app
 
 import (
+	"encoding/json"
 	"fmt"
 	"net/http"
 	"time"
 
+	"github.com/mattermost/mattermost-server/v6/app/request"
 	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/i18n"
 	"github.com/mattermost/mattermost-server/v6/shared/mlog"
 )
 
+func (a *App) NotifySystemAdminsToUpgrade(c *request.Context, notifyingUserID, currentUserTeamID string) *model.AppError {
+	user, appErr := a.GetUser(notifyingUserID)
+	if appErr != nil {
+		return appErr
+	}
+
+	// check if already notified
+	sysVal, err := a.Srv().Store.System().GetByName("NOTIFIED_ADMIN_TO_UPGRADE")
+	if err != nil {
+		mlog.Error("Unable to get NOTIFIED_ADMIN_TO_UPGRADE", mlog.Err(err))
+	}
+
+	alprnu := &model.AlreadyCloudNotifiedAdminUsersInfo{
+		Info: make([]model.UserInfo, 0),
+	}
+
+	if sysVal != nil {
+		val := sysVal.Value
+
+		err = json.Unmarshal([]byte(val), alprnu)
+		if err != nil {
+			mlog.Error("Unable to Unmarshal", mlog.Err(err))
+		}
+
+		if alprnu.ContainsID(user.Id) {
+			return model.NewAppError("app.SendCloudUpgradeConfirmationEmail", "api.cloud.notify_admin_to_upgrade_error.already_notified", nil, "", http.StatusForbidden)
```

Comment:
Can you please share if we have any requirements on this behaviour?

Reply:
Somewhere on the ticket for this PR it mentions `current thought is to just allow the admin to receive it once only`. @marianunez thoughts on this?